flink/src/test/java/org/apache/iceberg/flink/TestTableLoader.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink;
+
+import java.io.IOException;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestTableLoader {
+
+  @Test
+  public void testJavaSerialization() throws IOException, ClassNotFoundException {
+    TableLoader loader = TableLoader.fromHadoopTable("/my_path/my_table");
+    TableLoader copied = TestCatalogLoader.javaSerAndDeSer(loader);
+    Assert.assertEquals("HadoopTableLoader{location=/my_path/my_table}", copied.toString());
```

Comment:
I don't think it is sufficient to test that the loader's `toString` representation is the same. This would still pass if the underlying `Configuration` were just set to null and not serialized at all. I think it would make sense to do something that will fail if the `Configuration` is not serialized and deserialized correctly, like load the table. In that case, you would just need to create a Hadoop table on the local FS and load it through the copied loader.

Reply:
Agree